apps/nextjs-app/src/features/app/components/setting/account/AddPassword.tsx

```diff
@@ -0,0 +1,118 @@
+import { useMutation } from '@tanstack/react-query';
+import { addPassword } from '@teable/openapi';
+import { passwordSchema } from '@teable/openapi/src/auth/types';
+import { useSession } from '@teable/sdk/hooks';
+import { Spin } from '@teable/ui-lib/base';
+import {
+  Button,
+  Dialog,
+  DialogContent,
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+  DialogTrigger,
+  Input,
+  Label,
+  useToast,
+} from '@teable/ui-lib/shadcn';
+import { useTranslation } from 'next-i18next';
+import { useEffect, useState } from 'react';
+import { Error } from '@/components/Error';
+
+export const AddPassword = () => {
+  const [newPassword, setNewPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const { t } = useTranslation('common');
+  const [error, setError] = useState('');
+  const [open, setOpen] = useState(false);
+  const { toast } = useToast();
+  const { refresh } = useSession();
+
+  const { mutateAsync: addPasswordMutate, isLoading } = useMutation({
+    mutationFn: addPassword,
+    onSuccess: () => {
+      toast({ title: t('settings.account.addPasswordSuccess.title') });
+      setOpen(false);
+      refresh();
+    },
+  });
+
+  useEffect(() => {
+    if (newPassword && confirmPassword && newPassword !== confirmPassword) {
```

Comment:
In my sight, it's better to use different error tip to distinguish when newPassword or confirmPassword  is empty.

Reply:
I agree with you, differentiating error messages for newPassword and confirmPassword when they are empty would be beneficial. However, it's not important, I don't want to spend time on it right now.